pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -198,12 +198,112 @@ func (c *Cluster) validateOSDSettings() error {
 	return nil
 }
 
+func (c *Cluster) ValidateTopologyAcrossNodes() error {
+	nodes, err := k8sutil.GetKubernetesNodesMatchingRookNodes(c.clusterInfo.Context, c.spec.Storage.Nodes, c.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to list nodes for topology validation")
+	}
+
+	return checkTopologyConflicts(nodes)
+}
+
+func checkTopologyConflicts(nodes []corev1.Node) error {
+	hierarchy := []string{
```

Comment:
The topology label hierarchy actually can be found by calling the method `allKubernetesTopologyLabelsOrdered()`